Linux Trashing a symlink to a directory will trash the target directory not the symlink

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Usually, editors like VSCode will only delete the symlink itself, not the target's content or the target itself. I'm unsure what exactly gets moved to trash. but as a user, I'll just assume the symlink's target.

### Zed Version and System Specs

Zed: v0.166.1 (Zed)
OS: Linux X11 pop 22.04
Memory: 62.7 GiB
Architecture: x86_64
GPU: NVIDIA GeForce RTX 3060 || NVIDIA || 560.35.03

### If applicable, add screenshots or screencasts of the incorrect state / behavior

[Screencast from 12-24-2024 05:28:54 PM.webm](https://github.com/user-attachments/assets/7f1ff585-f829-4325-ad6f-baa380294725)

notice the /simple/src folder disappearing. it got moved to trash.

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary>

<!-- Click below this line and paste or drag-and-drop your log-->
```
2024-12-24T17:28:38.275163625+01:00 [INFO] ========== starting zed ==========
2024-12-24T17:28:38.297273596+01:00 [INFO] Compositor detection: _NET_WM_CM_S?=false, _NET_WM_CM_OWNER=false, _NET_SUPPORTING_WM_CHECK=true
2024-12-24T17:28:38.297392311+01:00 [INFO] x11: compositor present: true, gtk_frame_extents_supported: true
2024-12-24T17:28:38.304628084+01:00 [INFO] Try connect ibus
2024-12-24T17:28:38.305411814+01:00 [INFO] perform;
2024-12-24T17:28:38.305475534+01:00 [INFO] authenticate;
2024-12-24T17:28:38.305517694+01:00 [INFO] Opening main db
2024-12-24T17:28:38.305587005+01:00 [INFO] read_command;
2024-12-24T17:28:38.305655815+01:00 [INFO] send_secondary_commands;
2024-12-24T17:28:38.306063355+01:00 [INFO] new;
2024-12-24T17:28:38.306125051+01:00 [INFO] socket reader;
2024-12-24T17:28:38.306544804+01:00 [INFO] Opening main db
2024-12-24T17:28:38.306718021+01:00 [INFO] keep_updated;
2024-12-24T17:28:38.308731262+01:00 [INFO] Using git binary path: None
2024-12-24T17:28:38.34654109+01:00 [INFO] set environment variables from shell:/bin/bash, path:/home/robin/.local/bin:/home/robin/.local/bin:/home/robin/.cargo/bin:/home/robin/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/robin/.local/bin
2024-12-24T17:28:38.346757088+01:00 [INFO] No prompt template overrides directory found at /home/robin/.local/share/zed/prompt_overrides. Using built-in prompts.
2024-12-24T17:28:38.359352382+01:00 [INFO] extensions updated. loading 3, reloading 0, unloading 0
2024-12-24T17:28:38.361617309+01:00 [INFO] activate is not implemented on Linux, ignoring the call
2024-12-24T17:28:38.362826623+01:00 [INFO] Opening main db
2024-12-24T17:28:38.363189007+01:00 [INFO] perform;
2024-12-24T17:28:38.363239903+01:00 [INFO] authenticate;
2024-12-24T17:28:38.363298835+01:00 [INFO] read_command;
2024-12-24T17:28:38.363674083+01:00 [INFO] socket reader;
2024-12-24T17:28:38.367409888+01:00 [INFO] Using Visual { id: 35, colormap: 0, depth: 32 }
2024-12-24T17:28:38.367463038+01:00 [INFO] Creating colormap 41943043
2024-12-24T17:28:38.369699491+01:00 [INFO] new;
2024-12-24T17:28:38.370411235+01:00 [INFO] keep_updated;
2024-12-24T17:28:38.407824875+01:00 [INFO] Enabling Vulkan Portability
2024-12-24T17:28:38.407863739+01:00 [INFO] Enabling color space support
2024-12-24T17:28:38.53860889+01:00 [INFO] Adapter: "NVIDIA GeForce RTX 3060"
2024-12-24T17:28:38.54363976+01:00 [INFO] Ray tracing is supported
2024-12-24T17:28:38.761487943+01:00 [INFO] Using surface present mode FIFO_RELAXED
2024-12-24T17:28:38.761734188+01:00 [WARN] Unable to forbid exclusive full screen
2024-12-24T17:28:38.824620429+01:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-12-24T17:28:38.824670603+01:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-12-24T17:28:38.82470574+01:00 [INFO] Initializing Blade pipelines for surface SurfaceInfo { format: Bgra8UnormSrgb, alpha: Ignored }
2024-12-24T17:28:38.87123857+01:00 [INFO] Refreshing at 10001 micros
2024-12-24T17:28:38.872502376+01:00 [INFO] Using GPU: GPUSpecs { is_software_emulated: false, device_name: "NVIDIA GeForce RTX 3060", driver_name: "NVIDIA", driver_info: "560.35.03" }
2024-12-24T17:28:38.893334705+01:00 [WARN] GPU frame is out of date
2024-12-24T17:28:38.893389698+01:00 [INFO] XConnected server on 31457287, transport version: 0.0, TRANSPORT_MAX: 20
2024-12-24T17:28:38.894491369+01:00 [INFO] Using surface present mode FIFO_RELAXED
2024-12-24T17:28:38.894841801+01:00 [WARN] Unable to forbid exclusive full screen
2024-12-24T17:28:38.900881185+01:00 [INFO] set status on client 0: Authenticating
2024-12-24T17:28:38.90137609+01:00 [INFO] perform;
2024-12-24T17:28:38.901432376+01:00 [INFO] authenticate;
2024-12-24T17:28:38.901485336+01:00 [INFO] read_command;
2024-12-24T17:28:38.901968649+01:00 [INFO] socket reader;
2024-12-24T17:28:38.907570457+01:00 [INFO] new;
2024-12-24T17:28:38.908289354+01:00 [INFO] keep_updated;
2024-12-24T17:28:38.919860685+01:00 [ERROR] missing field `name` at line 20 column 1
2024-12-24T17:28:38.92173781+01:00 [INFO] set status on client 84084: Connecting
2024-12-24T17:28:38.964250869+01:00 [INFO] Opening main db
2024-12-24T17:28:38.973939955+01:00 [INFO] using project environment variables from CLI. PATH="/home/robin/.local/bin:/home/robin/.cargo/bin:/home/robin/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin"
2024-12-24T17:28:39.132579531+01:00 [INFO] connected to rpc endpoint https://collab.zed.dev/rpc
2024-12-24T17:28:39.497615214+01:00 [INFO] add_connection;
2024-12-24T17:28:39.509736803+01:00 [INFO] set status on client 84084: Connected { peer_id: PeerId { owner_id: 629, id: 4889042 }, connection_id: ConnectionId { owner_id: 0, id: 0 } }
2024-12-24T17:29:01.946733242+01:00 [INFO] new;
2024-12-24T17:29:01.947411393+01:00 [INFO] keep_updated;
2024-12-24T17:29:01.948831334+01:00 [ERROR] error reading file "/home/robin/Desktop/simple/build/src" on event: No such file or directory (os error 2)
2024-12-24T17:29:02.049221545+01:00 [ERROR] error reading file "/home/robin/Desktop/simple/build/src" on event: No such file or directory (os error 2)
```
<!-- Click above this line and paste or drag-and-drop your log--></details>

--- a/crates/fs/src/fs.rs
+++ b/crates/fs/src/fs.rs
@@ -11,8 +11,7 @@ use git::GitHostingProviderRegistry;
 use ashpd::desktop::trash;
 #[cfg(any(target_os = "linux", target_os = "freebsd"))]
 use smol::process::Command;
-#[cfg(any(target_os = "linux", target_os = "freebsd"))]
-use std::fs::File;
+
 #[cfg(unix)]
 use std::os::fd::AsFd;
 #[cfg(unix)]
@@ -445,7 +444,13 @@ impl Fs for RealFs {
 
     #[cfg(any(target_os = "linux", target_os = "freebsd"))]
     async fn trash_file(&self, path: &Path, _options: RemoveOptions) -> Result<()> {
-        let file = File::open(path)?;
+        if let Ok(Some(metadata)) = self.metadata(path).await {
+            if metadata.is_symlink {
+                // TODO: trash_file does not support trashing symlinks yet - https://github.com/bilelmoussaoui/ashpd/issues/255
+                return self.remove_file(path, RemoveOptions::default()).await;
+            }
+        }
+        let file = smol::fs::File::open(path).await?;
         match trash::trash_file(&file.as_fd()).await {
             Ok(_) => Ok(()),
             Err(err) => Err(anyhow::Error::new(err)),

